### ISSUE TITLE
godot_videodecoder_open_file is modifying input_format struct owned by system.

### ISSUE THREAD

**Issue by ecurtz** [kidrigger/godot-videodecoder#47]
Although technically it isn't const  this is causing a crash on OS X and is probably a bad idea in general. Also the pointer doesn't appear to actually be used after the flags are modified, so it's probably safe to not do this?

https://github.com/kidrigger/godot-videodecoder/blob/af71383dce66eea88f3ce99b416613b925be3ba0/src/gdnative_videodecoder.c#L552

**Comment by jamie-pate** [comment]
I agree that overriding the output of `probe` is probably a mistake.
